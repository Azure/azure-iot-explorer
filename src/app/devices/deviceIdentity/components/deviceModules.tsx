/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { Route, useRouteMatch, useLocation } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { ModuleIdentityTwin } from '../../module/moduleIdentityTwin/components/moduleIdentityTwin';
import { AddModuleIdentity } from '../../module/addModuleIdentity/components/addModuleIdentity';
import { ModuleIdentityList } from '../../module/moduleIdentityList/components/moduleIdentityList';
import { ModuleIdentityDetail } from '../../module/moduleIndentityDetail/components/moduleIdentityDetail';
import { ModuleDirectMethod } from '../../module/moduleDirectMethod/components/moduleDirectMethod';
import { BreadcrumbRoute } from '../../../navigation/components/breadcrumbRoute';
import { ROUTE_PARTS } from '../../../constants/routes';
import { ResourceKeys } from '../../../../localization/resourceKeys';
import { getModuleIdentityIdFromQueryString } from '../../../shared/utils/queryStringHelper';

export const DeviceModules: React.FC = () => {
    const { url } = useRouteMatch();
    const { t } = useTranslation();
    const { search } = useLocation();
    const moduleId = getModuleIdentityIdFromQueryString(search);

    return (
        <>
            <Route exact={true} path={`${url}`} component={ModuleIdentityList}/>
            <BreadcrumbRoute
                exact={true}
                path={`${url}/${ROUTE_PARTS.ADD}`}
                breadcrumb={{name: t(ResourceKeys.breadcrumb.addModuleIdentity)}}
                children={<AddModuleIdentity/>}
            />

            <BreadcrumbRoute
                exact={true}
                path={`${url}/${ROUTE_PARTS.MODULE_DETAIL}`}
                breadcrumb={{name: moduleId, suffix: search}}
                children={<ModuleIdentityDetail/>}
            />

            <BreadcrumbRoute
                exact={true}
                path={`${url}/${ROUTE_PARTS.MODULE_TWIN}`}
                breadcrumb={{name: moduleId, suffix: search}}
                children={<ModuleIdentityTwin/>}
            />

            <BreadcrumbRoute
                exact={true}
                path={`${url}/${ROUTE_PARTS.MODULE_METHOD}`}
                breadcrumb={{name: moduleId, suffix: search}}
                children={<ModuleDirectMethod/>}
            />
        </>
    );
};
