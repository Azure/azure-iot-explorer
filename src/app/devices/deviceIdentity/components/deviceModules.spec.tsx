/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { shallow } from 'enzyme';
import { DeviceModules } from './deviceModules';

jest.mock('react-router-dom', () => ({
    useLocation: () => ({ search: '?moduleId=mod1' }),
    useRouteMatch: () => ({ url: '' })
}));


describe('DeviceModules', () => {
    it('matches snapshot', () => {
        expect(shallow(
            <DeviceModules/>
        )).toMatchSnapshot();
    });
});