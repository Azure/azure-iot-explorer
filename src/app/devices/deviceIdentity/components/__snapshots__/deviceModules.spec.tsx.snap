// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceModules matches snapshot 1`] = `
<Fragment>
  <Component
    component={[Function]}
    exact={true}
    path=""
  />
  <Component
    breadcrumb={
      Object {
        "name": "breadcrumb.addModuleIdentity",
      }
    }
    exact={true}
    path="/add"
  >
    <Component />
  </Component>
  <Component
    breadcrumb={
      Object {
        "name": "mod1",
        "suffix": "?moduleId=mod1",
      }
    }
    exact={true}
    path="/moduleDetail"
  >
    <Component />
  </Component>
  <Component
    breadcrumb={
      Object {
        "name": "mod1",
        "suffix": "?moduleId=mod1",
      }
    }
    exact={true}
    path="/moduleTwin"
  >
    <Component />
  </Component>
  <Component
    breadcrumb={
      Object {
        "name": "mod1",
        "suffix": "?moduleId=mod1",
      }
    }
    exact={true}
    path="/moduleMethod"
  >
    <Component />
  </Component>
</Fragment>
`;
